lib/jarvis/defer.rb

```diff
@@ -0,0 +1,15 @@
+module Jarvis class Defer
+  def initialize
+    @deferred = []
+  end
+
+  def do(&block)
+    @deferred << block
+  end
+
+  def run
+    @deferred.each do |block|
+      block.call
+    end
+  end
+end end
```

Comment:
Yeah, you can always check for their execution status,

http://ruby-concurrency.github.io/concurrent-ruby/Concurrent/Future.html